src/main/java/de/uni_tuebingen/ub/nppm/servlet/RESTServlet.java

```diff
@@ -0,0 +1,94 @@
+package de.uni_tuebingen.ub.nppm.servlet;
+
+import de.uni_tuebingen.ub.nppm.db.EditionDB;
+import de.uni_tuebingen.ub.nppm.db.EinzelbelegDB;
+import de.uni_tuebingen.ub.nppm.db.MghLemmaDB;
+import de.uni_tuebingen.ub.nppm.db.NamenKommentarDB;
+import de.uni_tuebingen.ub.nppm.model.Edition;
+import de.uni_tuebingen.ub.nppm.model.Einzelbeleg;
+import de.uni_tuebingen.ub.nppm.model.MghLemma;
+import de.uni_tuebingen.ub.nppm.model.NamenKommentar;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.json.*;
+/*
+    Example of REST implementation with json output
+    Call URL -> http://localhost:8080/neg/rest?name=test&entity=mghlemma
+    Call URL -> http://localhost:8080/neg/rest?name=test&entity=namenkommentar
+ */
+public class RESTServlet extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        try {   
+            response.setContentType("application/json");
+            response.setCharacterEncoding("UTF-8");
+            String name = request.getParameter("name");
+            String entity = request.getParameter("entity");
+            if(name == null || name.isEmpty()){
+                response.getWriter().write("Please specify name parameter");
+            }
+            else if(entity != null && entity.compareTo("mghlemma") == 0) {
+                List<MghLemma> lemmas = null;                
+                try {
+                    lemmas = MghLemmaDB.getByName(name);
+                } catch (Exception ex) {
+                    response.getWriter().write(ex.getLocalizedMessage());                    
+                }
+
+                if (lemmas.size() > 0) {
+                    try {
+                        String json = "\n{\n";
+                        json += " [\n";
+                        for(MghLemma lemma: lemmas){
+                            json += "  {\n";
```

Comment:
JSON results should not be built as strings, but by using a proper org.json object model instead. Please have a look at the AjaxServlet for examples.

Reply:
Also, there will be multiple use cases for getting the JSON representation of a certain object, so it would be better to move the logic into a getJSON() function in the corresponding model class.

Reply:
I have created the function getJSON in the Model Classes